src/commonMain/kotlin/exchange.dydx.abacus/state/manager/V4StateManagerAdaptor.kt

```diff
@@ -295,9 +296,18 @@ class V4StateManagerAdaptor(
     }
 
     private fun getRewardsParams() {
-        getOnChain(QueryType.RewardsParams, null) { response ->
+        getOnChain(QueryType.RewardsParams, null) { rewardsParams ->
             val oldState = stateMachine.state
-            update(stateMachine.onChainRewardsParams(response), oldState)
+            update(stateMachine.onChainRewardsParams(rewardsParams), oldState)
+
+            val json = Json.parseToJsonElement(rewardsParams).jsonObject.toMap()
+            val marketId = parser.asString(parser.value(json, "params.marketId"))
+            val params = iMapOf("marketId" to marketId)
+            val paramsInJson = jsonEncoder.encode(params)
+
+            getOnChain(QueryType.GetMarketPrice, paramsInJson) { marketPrice ->
```

Comment:
only fetch once for now, but should change to fetch based on timestamp on calculation